### ISSUE TITLE
A way to retrieve the validated data

### ISSUE THREAD

**Issue by arthurfranca** [mikeerickson/validatorjs#307]
Sometimes the input data object from a request has many fields that aren't allowed.
Is there a way to omit those fields not listed on rules object and retrieve the remaining ones after running validation?

Something like:
```javascript
if (validation.passes()) {
  // Input data object clone without fields that aren't rule keys
  permittedParams = validation.output 
  Model.create(permittedParams)
} else {
  // An object with fields from succesful rules (it would allow using the valid portion of the data)
  // Empty object when any required field was missing
  console.log(validation.output)
}
```

On [Laravel docs](https://laravel.com/docs/5.7/validation) it seems to be possible with
```php
$validated = $request->validated()
```

**Comment by danialdezfouli** [comment]
```
function getNestedData(input, attribute) {
  let key = attribute;
  let value = input;
  const exploded = attribute.split('.');
  const length = Math.min(exploded.length, 10);

  for (let i = 0; i < length; i++) {
    key = exploded[i];
    if (!(key in value)) {
      key = false;
      break;
    }
    value = value[key];
  }

  return [key, value];
}

export function validatedData(validator: Validator) {
  const { input, rules } = validator;
  const values = new Map();

  for (const attribute in rules) {
    if (typeof input[attribute] != 'undefined') {
      values[attribute] = input[attribute];
    } else {
      if (attribute.includes('.')) {
        const [key, value] = getNestedData(input, attribute);
        if (key !== false) {
          values[attribute] = value;
        }
      }
    }
  }

  return values;
}
```

**Comment by mikeerickson** [comment]
Added in v3.5 using `successes` method

**Comment by scube-elate** [comment]
@mikeerickson is `successes` released already? I couldn't find the method on v3.5.

**Comment by cednore** [comment]
@mikeerickson I've opened a new PR addressing this issue. https://github.com/mikeerickson/validatorjs/pull/444
